src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -450,61 +450,69 @@ internal async Task<bool> ReceiveJSDataChunk(long streamId, long chunkId, byte[]
             }
         }
 
-        public async Task SendDotNetStreamAsync(long streamId, ChannelWriter<ArraySegment<byte>> writer)
+        public async Task<int> SendDotNetStreamAsync(DotNetStreamReference dotNetStreamReference, byte[] buffer)
```

Comment:
Note, this last commit https://github.com/dotnet/aspnetcore/pull/34817/commits/1a8bc1d7e61a42a8255ec623ea9f63458a69e213 is a _rough_ draft of an `IAsyncEnumerable` based approach to SignalR streams, to avoid issues I'm seeing with using ArrayPools in conjunction with `Channels`. If we decide to pursue this approach, it'll warrant some cleanup.